src/Github/Api/V3/CreatePullRequestThroughApiCall.php

```diff
@@ -63,16 +62,14 @@ public function __invoke(
 
         $response = $this->client->sendRequest($request);
 
+        Psl\invariant($response->getStatusCode() >= 200 || $response->getStatusCode() <= 299, 'Failed to create pull request through GitHub API.');
+
         $responseBody = $response
             ->getBody()
             ->__toString();
 
-        Assert::greaterThanEq($response->getStatusCode(), 200);
-        Assert::lessThanEq($response->getStatusCode(), 299);
-
-        $responseData = json_decode($responseBody, true);
-
-        Assert::isMap($responseData);
-        Assert::keyExists($responseData, 'url');
+        Json\typed($responseBody, Type\shape([
+            'url' => Type\mixed(),
+        ], allow_unknown_fields: true));
```

Comment:
please avoid this - only confusing, would prefer just the boolean

Reply:
hm, actually this doesn't need to be set to true;